src/session/content/message_row/mod.rs

```diff
@@ -2,149 +2,307 @@ mod sticker;
 mod sticker_paintable;
 mod text;
 
-use self::sticker::MessageSticker;
+pub use self::sticker::MessageSticker;
 use self::sticker_paintable::StickerPaintable;
-use self::text::MessageText;
+pub use self::text::MessageText;
```

Comment:
Is this mixing of public and private usages intentional?
If it is, just ignore this.

Another approach for better readability and better looking would be to separate public from private imports (personal taste).

Reply:
I prefer keeping them like this so that they're ordered by module name (and to avoid creating another group).